app/controllers/system_admin/funding_controller.rb

```diff
@@ -40,12 +40,16 @@ def funding_path(year, funding_type)
         send("provider_funding_#{funding_type}_path", provider_id: organisation.id, academic_year: year)
       when School
         send("lead_school_funding_#{funding_type}_path", lead_school_id: organisation.id, academic_year: year)
+      when LeadPartner
+        send("lead_partner_funding_#{funding_type}_path", lead_partner_id: organisation.id, academic_year: year)
       end
     end
 
     def organisation
       return Provider.find(params[:provider_id]) if params[:provider_id].present?
 
+      return LeadPartner.find(params[:lead_partner_id]) if params[:lead_partner_id].present?
+
       School.find(params[:lead_school_id])
     end
```

Comment:
This looks similar to [app/controllers/system_admin/funding/trainee_summaries_controller.rb:40](https://github.com/DFE-Digital/register-trainee-teachers/pull/4368/files#diff-c129a7f35b65726410d36997e4d3caa083781e7ed77e2ac2012c31c0d28c3278R40) could this controller inherit from `::Funding::BaseFundingController` or extract this to a concern?